build-charm

```diff
@@ -52,7 +52,14 @@ if grep "^\[testenv:build\]$" $charm_dir/tox.ini &> /dev/null; then
   echo " . Building $charm_dir ($charm_name) via tox"
   DIR="$(pwd)"
   cd $charm_dir
-  tox -e build
+  # Let us retry a couple of times in case of a transient failure fetching bits
+  n=0
+  until [ $n -ge 3 ]
+  do
+    tox -e build && break
+    n=$[$n+1]
+    sleep 3
```

Comment:
I'd rather see an exponential backoff here (3,9,27) instead of a straight up retry.

If the root cause of this issue is resource exhaustion, we're only going to make it worse if we slam the build servers with a quick succession of retries ...

Reply:
Agree, this would be an improvement over the flat delay, and I'd be in support of a follow-on change to do that.

Reply:
https://github.com/openstack-charmers/release-tools/pull/54 has been opened to resolve this request, adding a general purpose retry function that uses exponential backoff for the retries!